### PR TITLE
subscribe publish and JetStream options

### PR DESCRIPTION
**Pull request by scottf**
also fix String to string

### CODE REVIEW

**Review by ColinSullivan1 — APPROVED** [review]
(edited.). I think the publish options are duplicates / replacements of code already in the branch.

File: src/NATS.Client/JetStream/PublishOptions.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 The NATS Authors
```
Comment (ColinSullivan1) [inline src/NATS.Client/JetStream/PublishOptions.cs:1]: Weren't these already pushed to the JS Branch?

Comment (ColinSullivan1) [reply src/NATS.Client/JetStream/PublishOptions.cs:1]: e.g https://github.com/nats-io/nats.net/commit/1bc70fc20fd360d280d09601230478789a05489d

Comment (scottf) [reply src/NATS.Client/JetStream/PublishOptions.cs:1]: My mistake. This is all fixed. I got the right version and applied my changes.

**Comment by ColinSullivan1** [comment]
Looks like the TestPublishOptions.cs unit test got deleted...

**Comment by scottf** [comment]
> Looks like the TestPublishOptions.cs unit test got deleted...

It was moved into the JetStream namespace folder

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.

**Pull request merged**